### PR TITLE
incorporated @aurelw 's pull request of updating car, also added the …

### PR DESCRIPTION
**Pull request by qiang-bi**
…feature that if gti push, the screen shows a man pushing a car

### CODE REVIEW

File: gti.c [hunk 1]
```diff
@@ -72,14 +72,18 @@ int main(int argc, char **argv)
...
-    SLEEP_DELAY = 1000000 / (TERM_WIDTH + GTI_SPEED);
-
+    SLEEP_DELAY = 1300000 / (TERM_WIDTH + GTI_SPEED);
+   
     init_space();
-    for (i = -20; i < TERM_WIDTH; i++) {
-        draw_car(i);
+	for (i = -20; i < TERM_WIDTH; i++) {
+      if (!strcmp(push, "push")) push_car(i); 
+       else draw_car(i);
```
Comment (rwos) [inline gti.c]: Ha, so, the `push_car()` stuff is nice, but:
* why not strcmp with `argv[1]` directly? (but check argc before access!)
* the indentation here needs fixing
* don't change the `#defines` or the `SLEEP_DELAY`, or other unrelated stuff
* don't commit the binary
* I still don't understand the "opel" (or am I reading this wrong?). This is *clearly* a VW Golf GTI Mark 1. It even says so on the car. :D

**Comment by qiang-bi** [comment]
Changes made accordingly. Please review again

**Comment by coladict** [comment]
Maybe it should have a check for pre-command options like these https://www.git-scm.com/docs/git
While it starts with a dash, it's a pre-command option, so don't check for push yet.

**Comment by qiang-bi** [comment]
Do you think its ok to just search for whether the word "push" exists in argv? Or should I check each element in argv and pushcar only if argv contains only  -x --x and "push" ?

**Comment by coladict** [comment]
I was thinking more the lines of 
```C
for (int i =1; i < argc; ++i) {
  if (argv[i][0] == '-') continue;
  if (!strcmp(argv[i], "push")) return 1;
  break;
}
return 0;
```

**Comment by qiang-bi** [comment]
ok i have just updated the function

**Comment by TomK32** [comment]
Can't we have the guy pushing in a separate commit? Would like to have that for TomK32/popel

File: gti.c [hunk 2]
```diff
@@ -185,15 +199,44 @@ void line_at(int start_x, const char *s)
...
+    if (x == 23) {
+        line_at(x, "                 | |");
+        line_at(x, "  ___  _ __   ___| |");
+        line_at(x, " / _ \\| '_ \\ / _ \\ |");
+        line_at(x, "| (_) | |_) |  __/ |");
+        line_at(x, " \\___/| .__/ \\___|_|");
+        line_at(x, "      | |           ");
+        line_at(x, "      |_|           ");
+        return;
+    }
```
Comment (rwos) [inline gti.c]: nope, I don't want this...

File: gti.c [hunk 3]
```diff
@@ -207,8 +250,13 @@ void draw_car(int x)
     line_at(x, " `   X   --------------   X   '");
     line_at(x, "   ':-:'                ':-:'  ");
     }
+
 }
 
+
+
+
+
```
Comment (rwos) [inline gti.c]: please remove the empty lines again

File: gti.c [hunk 4]
```diff
@@ -96,6 +100,16 @@ int main(int argc, char **argv)
...
 
+/*return true if push command found*/
+int checkPushCommand(int argc, char** argv) {
+  for (int i = 1; i < argc; i++) {
+    if (argv[i][0] == '-') continue;
+    if (!strcmp(argv[i], "push")) return 1;
+    break;
+  }
+  return 0;
+}
```
Comment (rwos) [inline gti.c]: So you're checking/strcmp'ing this every frame - would be nicer to do this only once at the top of main() and set a variable. Also, remove the `(void) argc;` cast (line 75), that's just to silence the "argument unused" warning, and you're using it now :)

code style nitpicking:
* I'm using 4 spaces for indentation
* the `{` after the function definition should be on a new line
* use snake_case names, not camelCase
* move the `**` to the variable (`char **argv`, not `char** argv`)

**Comment by qiang-bi** [comment]
Just pushed a revision. Thanks for pointing out my ugly code style...

File: gti.c [hunk 5]
```diff
@@ -71,17 +72,20 @@ int main(int argc, char **argv)
...
     init_space();
     for (i = -20; i < TERM_WIDTH; i++) {
+      if (pushing)
+        push_car(i);
+      else
         draw_car(i);
-        usleep(SLEEP_DELAY);
-        clear_car(i);
+      usleep(SLEEP_DELAY);
+      clear_car(i);
```
Comment (rwos) [inline gti.c]: This is still using two spaces ;) but I'll add an `indent` script or somthing to the Makefile. Okay for now.

**Comment by rwos** [comment]
@qiang-bi thank you!
